src/cbmc/bmc.cpp

```diff
@@ -45,6 +45,11 @@ void bmct::do_unwind_module()
   // this is a hook for hw-cbmc
 }
 
+void bmct::freeze_program_variables()
```

Comment:
Please add some comments to indicate (a) at a high level what will happen here in the CEGIS case, and (b) how / why would some other program use this hook?

Reply:
Detailed enough?:
~~~
/// Hook used by CEGIS to selectively freeze variables
/// in the SAT solver after the SSA formula is added to the solver.
/// Freezing variables is necessary to make use of incremental
/// solving with MiniSat SimpSolver.
/// Potentially a useful hook for other applications using
/// incremental solving.
~~~

Reply:
Also, I don't know who maintains hw-cbmc, but maybe a future change could be to have a couple of generic hook points, one before converting to SSA, and one after. hw-cbmc currently has "do_unwind_modules" before converting to SSA.

Reply:
Right, exactly. We should have hooks that people can re-use. the description looks good to me.